docs/rules/sort-comp.md

```diff
@@ -171,6 +172,48 @@ var Hello = React.createClass({
 });
 ```
 
+If you want to flow annotations to be at the top:
+
+```js
+"react/sort-comp": [1, {
+  order: [
+    'type-annotations',
+    'static-methods',
+    'lifecycle',
+    'everything-else',
+    'render',
+  ],
+}]
+```
+
+With the above configuration, the following patterns are considered warnings:
+
+```js
+class Hello extends React.Component<any, Props, void> {
+  onClick: function() { this._someElem = true; }
+  props: Props;
+  _someElem: bool;
+  render: function() {
```

Comment:
not a big deal, but `class` examples should be using concise method syntax - i'm actually not even sure it will work with the `:` syntax you've got here. (ie `onClick() {` not `onClick: function () {`